### DISCUSSION TITLE
Second deployed instance of Community Platform

### DISCUSSION THREAD

**Discussion by thisislawatts** [ONEARMY/community-platform#1254] [Feature request / Ideas]
**Proposed Timeline:**  December 13th, 2021
**Lead:** @thisislawatts 

### Rationale

The Community Platform is currently installed on one site, [community.preciousplastic.com](https://community.preciousplastic.com), there are two additional projects from One Army which would benefit from the some of the features and functionality of the Platform, these are https://projectkamp.com and https://fixing.fashion.

We have a number of Platform features under development which are not strictly required for some of the other projects. The goal of this work is to create a second deployed instance of the community platform with a minimal featureset, to support the efforts of Project Kamp particularly around gathering Research.

### Requirements

1. Version of Community Platform deployed to [community.projectkamp.com](https://community.projectkamp.com)
1. Limited subset of Community Platform functionality
     1. User sign up and login
     2. Research module 
     3. Existing academy content using same mechanism as Precious Plastic
1. Themeable design to align with project identity
1. Analytics events so we can understand how people are using the site
1. Automated deployments
1. Documentation covering release process

### Out of scope

The following items/functionality will not be delivered as part of this work.

* Comments
* Maps
* Events
* How tos

### Timeline

A short 6 week timeline;

**Week 1**
- [x] 🙈 [Clean development install of dev-community-projectkamp](https://github.com/ONEARMY/community-platform/pull/1255)
- [x] Browser title lines up
- [x] Academy content loaded from existing system

**Week 2** 
- [ ] 🛑 Themed instance of site – **Blocked** A PR already exists which introduces theming capabilities to the Platform https://github.com/ONEARMY/community-platform/pull/1242

**Week 3**
- [x] Instance with all features removed, except auth, Research and Academy - https://github.com/ONEARMY/community-platform/pull/1255/commits/c006e25281f63bacea1f223d113d9c9da10e5bb0
- [x] Profile page is simplified to remove `Focus` section - https://github.com/ONEARMY/community-platform/pull/1255/commits/c006e25281f63bacea1f223d113d9c9da10e5bb0
- [x] Mobile/Desktop menu only show links relevant to current site 

**Week 4**
- [ ] Automated deploys to dev-community-projectkamp via primary platform deployment pipeline (Circle CI)

**Week 5**
- [ ] 🛑  Automated deploymentss to production install – **Blocked** This relies on us being able to make a Production release of the site. There have been no deployments since August.

Week 6
- [ ] 🚀 Launch community.projectkamp.com

### Advantages

* Deploying a second instance of the site supports community growth around Project Kamp 
* Provides a template for deploying an additional version of Platform for Fixing Fashion.
* Additional deployments for the project are fairly commonplace already with the additional development environments.

### Drawbacks

* Could make feature development more complicated as we will need to ensure functionality intended for one install doesn't "leak" into other minimal instances.

**Comment by thisislawatts** [comment]
🧇 Install available at:
https://la-project-kamp-development.web.app/

**Comment by davehakkens** [comment]
Exciting!

**Comment by thisislawatts** [comment]
Sharing here a proposal for how the release process should work for the additional Project Kamp environment:

![Screenshot 2021-11-13 at 10 27 45](https://user-images.githubusercontent.com/472589/141613691-9ed5a69f-7cea-498f-883e-93b90d3b3407.jpg)
https://whimsical.com/releasing-the-platform-to-production-environment-HQAFyzRVsBMxMdKYrauaA4

**Comment by thisislawatts** [comment]
Checklist:

* [ ] Analytics for the new production environment

**Comment by davehakkens** [comment]
> Checklist:
> 
>     * [ ]  Analytics for the new production environment

invited you on mail to an analytics account for this domain.. This was automatically setup when creating the firebase project.
Maybe we can use that?

**Comment by thisislawatts** [comment]
I've introduced a new label: [🐣 Second Instance](https://github.com/ONEARMY/community-platform/pulls?q=is%3Apr+is%3Aopen+label%3A%22%F0%9F%90%A3+Second+Instance%22) to track the PRs related to this discussion

**Comment by thisislawatts** [comment]
I will close out this discussion as we now have a second instance deployed:
https://community.projectkamp.com/
